presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PullUpExpressionInLambdaRules.java

```diff
@@ -326,9 +341,11 @@ public Boolean visitInputReference(InputReferenceExpression reference, Boolean c
         }
 
         // WHEN expression should only exist within SWITCH expression, and will throw exception in RowExpressionInterpreter, also no byte code generator for standalone WHEN expression
+        // Pull out LikePatternType and JoniRegexpType out can lead to byte code generation failure because of the underlying Regex type.
```

Comment:
Update the code, instead of filter out Regex.class, use an allowlist of boolean, long, double, slice, block, which will cover most common types, including integer, bigint, varchar, map, array, boolean, row types etc.

Reply:
But Joni regexp is a Slice in the backend right? Just making sure that usecase still works

Reply:
No, Joni regexp is has Regex as java type https://github.com/prestodb/presto/blob/ab71e549a5a1c9701188f6074b2b88745fcc69e8/presto-main/src/main/java/com/facebook/presto/type/JoniRegexpType.java#L35